gsa/src/web/pages/reports/detailspage.js

```diff
@@ -554,6 +590,20 @@ class ReportDetails extends React.Component {
             onCloseClick={this.handleFilterDialogClose}
           />
         }
+        {showDownloadReportDialog &&
+          <DownloadReportDialog
+            applyOverrides={applyOverrides}
+            filter={filter}
+            includeNotes={includeNotes}
+            includeOverrides={includeOverrides}
+            reportFormatId={reportFormatId}
+            reportFormats={reportFormats}
+            storeAsDefault={storeAsDefault}
+            onClose={this.handleCloseDownloadReportDialog}
+            onSave={this.handleReportDownload}
+            onValueChange={this.handleValueChange}
```

Comment:
I don't think the dialog should update the values in the state of the detailspage directly. This will cause unnecessary rerendering of the report details page. Instead use uncontrolled values in the dialog only update the state at the details page on save.

Reply:
It seems you are already doing this because you aren't using onValueChange in DownloadReportDialog. Please remove this line and the handleValueChange method then.